ProjectLighthouse/Database.cs

```diff
@@ -312,6 +317,40 @@ public async Task UnqueueLevel(int userId, Slot queuedSlot)
         await this.SaveChangesAsync();
     }
 
+    public async Task BlockUser(int userId, User blockedUser)
+    {
+        if (userId == blockedUser.UserId)
+        {
+            return;
+        }
+        
+        User? user = await this.Users.FirstOrDefaultAsync(u => u.UserId == userId);
+        
+        BlockedProfile blockedProfile = new()
+        {
+            User = user,
+            BlockedUser = blockedUser,
+        };
+
+        await this.BlockedProfiles.AddAsync(blockedProfile);
+
+        await this.SaveChangesAsync();
+    }
+
+    public void UnblockUser(int userId, User blockedUser)
+    {
+        if (userId == blockedUser.UserId) return;
+
+        this.BlockedProfiles.RemoveWhere(bp => bp.BlockedUser == blockedUser && bp.UserId == userId);
```

Comment:
I'd have to look at the generated SQL, but it might be more efficient to compare against the user's id rather than the foreign key field.

Reply:
Who knows, I'd have to look into that

Reply:
I've done that just in case